app/src/core/components/SubmitButton.tsx

```diff
@@ -1,30 +1,31 @@
-import { Button, ButtonTypeMap, CircularProgress, ExtendButtonBase } from '@mui/material';
-import { MouseEvent, PropsWithChildren } from 'react';
+import type { ButtonTypeMap } from '@mui/material';
+import LoadingButton from '@mui/lab/LoadingButton';
+import { MouseEvent, PropsWithChildren, useState } from 'react';
 import { animated, useSpring } from 'react-spring';
 
+import Result, { isOk, isErr } from 'true-myth/result';
+
 const wobbleVolume = 5;
-const AnimatedButton = animated(Button);
+const AnimatedButton = animated(LoadingButton);
 
-export interface SubmitButtonProps extends ExtendButtonBase<ButtonTypeMap<{}, 'button'>> {
+type MuiButtonProps = ButtonTypeMap['props'];
+export interface SubmitButtonProps extends MuiButtonProps {
     disableErrorState?: boolean;
     disableSuccessState?: boolean; // TODO
     iconSize?: number;
-    isLoading?: boolean;
-    onClick?: (event: MouseEvent<HTMLButtonElement, Event>) => Promise<unknown>;
+    onClick?: (event: MouseEvent<HTMLButtonElement, Event>) => Promise<Result<unknown, unknown>>;
+    type: 'submit';
```

Comment:
even tho this is called `SubmitButton` we should leave an escape value for when it needs to be a different type (i.e. if you have a secondary submit-y type button that needs a loading state/etc inside a `<form>`, which needs to be `type="button"` so it doesnt submit the form itself). 

But we should def default the `type` prop to be `submit`.